x-pack/docs/en/rest-api/security/clear-roles-cache.asciidoc

```diff
@@ -27,8 +27,11 @@ For more information about the native realm, see
 [[security-api-clear-role-cache-path-params]]
 ==== {api-path-parms-title}
 
-`name`::
-  (string) The name of the role. 
+`<roles>`::
+(Required, string)
+Comma-separated list of roles to evict from the native role cache. Supports
+wildcard (`*`) patterns. To evict all roles from the cache, use `*`.
```

Comment:
```suggestion
Comma-separated list of roles to evict from the native role cache.
To evict all roles from the cache, use `*`.
```

We don't actually support wildcards other than `*`

Reply:
The success just means the request is executed on the node. But it does not mean the roles are actually cleared. If a role is not found, it just gets silently skipped. Unfortunately, this response does not tell you exactly what roles are cleared or even just how many roles are cleared. We also don't have any API to show you the status (e.g. how many 
entries etc) of the cache. These are things that we should improve upon.

One way you can test it is to turn on trace logging for `authz` and authenticate using a user with the role before and after invocation of the clear-cache API. When the role is indeed successfully cleared, you should see a log line says `Building role from descriptors ... for role names ...` when you try to authenticate afterwards.

Reply:
Thanks for the clarification @ywangd.

Reply:
FYI, I raised https://github.com/elastic/elasticsearch/issues/70281 for things we discussed in this PR.

Reply:
Thanks @ywangd. Based on that issue, it looks like several of our API docs could use an update similar to this one. I'll work on expanding the scope of this PR to include those fixes as well. Thanks again!